0001-kaa-protocol/README.md

```diff
@@ -7,83 +7,96 @@ contributors: Alexey Gamov <agamov@cybervisiontech.com>
 ---
 
 ## Introduction
-
-The document describes general guidelines, principles, and requirements for **Kaa protocol**.
+The document describes the general requirements, principles, design and guidelines for the **Kaa Protocol** (KP).
+KP is a standard protocol designed for connecting clients and endpoints (colloquially - devices) to a Kaa server.
 
 ## Requirements and constraints
-
 - Clients must be able to connect to Kaa server without using an SDK and with minimum custom implementation.
-- Encrypted and unencrypted channels should be supported.
-- Protocol should work over MQTT and CoAP.
-- Protocol must be extensible to support future extensions.
+- Encrypted and unencrypted channels must be supported.
```

Comment:
Please add information of asynchronous kind of the KP

Another proposition to add minimalistic requirement to the KP's resource paths:
1. Define default values where applicable (kaa version, json format, etc).
2. Make default values optional to the resource path, e.g.:
_kaav1/app/ext/status/json_ is same as _kaa/app/ext/status_
3. Allow multi-version messaging via KP: provided a client communicates by kaav2 for some feature arised with this version, it still uses kaav1 for API which got available in kaav1.

This will:
1. Compact the resource path by omiting unnecessary parts (decrease network overhead where required)
2. Allow better management of API versions
3. Improve understanding of the API